runtime/src/bank.rs

```diff
@@ -4082,14 +4081,23 @@ impl Bank {
         message: &SanitizedMessage,
         lamports_per_signature: u64,
     ) -> u64 {
+        let transaction_meta = process_compute_budget_instruction(
+            message.program_instructions_iter(),
+            &self.feature_set,
+            Some(self.cluster_type()),
+        )
+        .unwrap_or_default();
```

Comment:
transaction meta data are bound to `transaction`, not to `message`. Bank is given `SanitizedMessage` here to provide fee, forced one more ComputeBudget::process_instructinos() call here. It can be refactored away separately.